python/src/fast.cpp

```diff
@@ -183,4 +188,102 @@ void init_fast(nb::module_& parent_module) {
         Returns:
           array: The quantized version of ``w``
       )pbdoc");
+
+  m.def(
+      "metal_kernel",
+      [](std::string name,
+         std::string source,
+         std::map<std::string, nb::handle> inputs_,
```

Comment:
I would use `ScalarOrArray` here (like the other ops) and use `to_array` to do the conversion. That will support everything except list which I think is a bit more consistent with out other operations.

Reply:
Sounds good to me.